v2/cli/init_command.go

```diff
@@ -0,0 +1,256 @@
+		if overwrite, err = checkFileOverwrite(scriptPath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				scriptPath,
+				scriptBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+
+		// Also generate a minimal package.json file
+		var packageBytes []byte
+		packageBytes, err = execTemplate(packageTemplate, fields)
+		if err != nil {
+			return err
+		}
+
+		packagePath := path.Join(".brigade", "package.json")
+		if overwrite, err = checkFileOverwrite(packagePath); err != nil {
+			return err
+		} else if overwrite {
+			if err = ioutil.WriteFile(
+				packagePath,
+				packageBytes,
+				0644,
+			); err != nil {
+				return err
+			}
+		}
+	}
+
+	// Generate the project.yaml file
+	var projectBytes []byte
+	projectBytes, err = execTemplate(projectTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Check if the .brigade/project.yaml path exists.
+	// Confirm overwrite if so.
+	projectPath := path.Join(".brigade", "project.yaml")
+	if overwrite, err = checkFileOverwrite(projectPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			projectPath,
+			projectBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var notesBytes []byte
+	notesBytes, err = execTemplate(notesTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a NOTES.txt file
+	notesPath := path.Join(".brigade", "NOTES.txt")
+	if overwrite, err = checkFileOverwrite(notesPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			notesPath,
+			notesBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	var secretsBytes []byte
+	secretsBytes, err = execTemplate(secretsTemplate, fields)
+	if err != nil {
+		return err
+	}
+
+	// Create a secrets.yaml file
+	secretsPath := path.Join(".brigade", "secrets.yaml")
+	if overwrite, err = checkFileOverwrite(secretsPath); err != nil {
+		return err
+	} else if overwrite {
+		if err = ioutil.WriteFile(
+			secretsPath,
+			secretsBytes,
+			0644,
+		); err != nil {
+			return err
+		}
+	}
+
+	err = addToGitignore(
+		"./.gitignore",
+		".brigade/secrets.yaml",
+		".brigade/node_modules",
+	)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// addToGitIgnore detects if .gitignore exists. If it does, add
+// the given paths to it. If not, create it and add the specified paths
+// to it unconditionally
+func addToGitignore(gitIgnorePath string, pathsToIgnore ...string) error {
+	file, err :=
+		os.OpenFile(gitIgnorePath, os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0644)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	for _, path := range pathsToIgnore {
+		if _, err =
+			file.WriteString(fmt.Sprintf("\n%s", path)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// canonicalizeLanguage checks if a given lagnuage is allowed. If it is,
+// shorten it to its file extension. Otherwise, return an error.
+func canonicalizeLanguage(language string) (string, error) {
```

Comment:
I know I suggested the name `canonicalizeLanguage`. After understanding this PR much better, what we're not merely converting non-canonical input to something canonical, what we're _really_ doing here is converting input (like ts / typescript / js / javascript) to a file extension.

Suggesting another rename for clarity: `fileExtensionForLanguage(...)`